services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -1474,7 +1484,94 @@ public void sendPortUpEvents(final IOFSwitch sw) throws SwitchOperationException
                 }
             }
         }
+    }
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PortConfigurationResponse configurePort(final PortConfigurationRequest request) 
+            throws SwitchOperationException {
+        DatapathId dpId = DatapathId.of(request.getSwitchId());
+        IOFSwitch sw = lookupSwitch(dpId);
+        
+        OFPortDesc ofPortDesc = getPort(sw, request.getPortNo());
+        logger.debug("Switch: {}, port: {}", sw, ofPortDesc);
+        
+        PortStatus newStatus = PortStatus.getPortStatus(request.getStatus());
+        if (newStatus != null) {
+            updatePortStatus(sw, ofPortDesc, newStatus);
+        }
+
+        if (request.getSpeed() > 0) {
+            updatePortSpeed(sw, ofPortDesc, request.getSpeed());
+        }
 
+        return new PortConfigurationResponse(request.getSwitchId(), request.getPortNo(), true);
+    }
+    
+    private void updatePortStatus(final IOFSwitch sw, final OFPortDesc ofPortDesc, final PortStatus portStatus) {
+        OFPortDesc.Builder builder = ofPortDesc.createBuilder();
+    
+        Set<OFPortConfig> configs = builder.getConfig() == null 
+                ? EnumSet.noneOf(OFPortConfig.class) : new HashSet<>(builder.getConfig());
+    
+        if (portStatus == PortStatus.UP) {
+            configs.remove(OFPortConfig.PORT_DOWN);
+        } else {
+            configs.add(OFPortConfig.PORT_DOWN);
+        }
+        builder.setConfig(configs);
+        
+        Set<OFPortConfig> portMask = EnumSet.noneOf(OFPortConfig.class);
+        portMask.add(OFPortConfig.PORT_DOWN);
+        
+        OFPortDesc updatedPort = builder.build();
+        
+        OFPortMod ofPortMod = sw.getOFFactory().buildPortMod().setConfig(configs)
+                .setPortNo(updatedPort.getPortNo())
+                .setHwAddr(updatedPort.getHwAddr())
+                .setMask(portMask)
+                .build();
+
+        logger.debug("oFPortMod request {}", ofPortMod);
+
+        boolean result = sw.write(ofPortMod);
```

Comment:
`true` response from sw.write does not mean that status was applied or even that it will be applied by switch. There is no simple way to verify that message was applied. Required tool is implemented as part of `flow pings` feature. But it not merged yet.

You can keep current implementation, but you must understand that `true` here mean "message is added into outgoing queue` and nothing else.

Reply:
Acknowledged